server/src/test/java/org/opensearch/index/compositeindex/datacube/startree/aggregators/StaticValueAggregatorTests.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.compositeindex.datacube.startree.aggregators;
+
+import org.apache.lucene.util.NumericUtils;
+import org.opensearch.index.compositeindex.datacube.startree.aggregators.numerictype.StarTreeNumericType;
+import org.opensearch.test.OpenSearchTestCase;
+
+public class StaticValueAggregatorTests extends OpenSearchTestCase {
+
+    // tests the extreme case where normal sum will lose precision
+    public void testKahanSummation() {
+        double[] numbers = { 1e-16, 1, -1e-16 };
+        double expected = 1;
+
+        // initializing our sum aggregator to derive exact sum using kahan summation
+        double aggregatedValue = getAggregatedValue(numbers);
+        assertEquals(expected, aggregatedValue, 0);
+
+        // assert kahan summation plain logic with our aggregated value
+        double actual = kahanSum(numbers);
+        assertEquals(actual, aggregatedValue, 0);
+
+        // assert that normal sum fails for this case
+        double normalSum = normalSum(numbers);
+        assertNotEquals(expected, normalSum, 0);
+        assertNotEquals(actual, normalSum, 0);
+        assertNotEquals(aggregatedValue, normalSum, 0);
+
+    }
+
+    private static double getAggregatedValue(double[] numbers) {
+        // explicitly took double to test for most precision
+        // hard to run similar tests for different data types dynamically as inputs and precision vary
+        SumValueAggregator aggregator = new SumValueAggregator(StarTreeNumericType.DOUBLE);
+        double aggregatedValue = aggregator.getInitialAggregatedValueForSegmentDocValue(NumericUtils.doubleToSortableLong(numbers[0]));
+        aggregatedValue = aggregator.mergeAggregatedValueAndSegmentValue(aggregatedValue, NumericUtils.doubleToSortableLong(numbers[1]));
+        aggregatedValue = aggregator.mergeAggregatedValueAndSegmentValue(aggregatedValue, NumericUtils.doubleToSortableLong(numbers[2]));
+        return aggregatedValue;
+    }
+
+    private double kahanSum(double[] numbers) {
```

Comment:
What value does this add? We already have a kahan summation abstraction which we can rely on?